CONTRIBUTING.md

```diff
@@ -1,109 +1,112 @@
 
-    ./scripts/check.py header tree
-    ./scripts/check.py header tree --fix
-```
+1. Review the [`LICENSE`](LICENSE) file. By contributing to Velox, you agree that your
+   contributions will be licensed under that LICENSE file. This step needs to be
+   completed only once.
 
-All the available formatting commands can be displayed by using
-`./scripts/check.py help`.
+1. Start a discussion by creating a Github issue, or asking on Slack (unless the change is trivial).
+   * This step helps you identify possible collaborators and reviewers.
+   * Does the change align with technical vision and project values?
+   * Will the change conflict with another change in progress? If so, work with others to minimize impact.
+   * Is this change large? If so, work with others to break into smaller steps.
 
-## Continuous Integration and CircleCI
+1. Review our coding style and best practices document in [`CODING_STYLE.md`](CODING_STYLE.md).
 
-Velox uses CircleCI as the continuous integration system, so please ensure your
-PR does not break any of these workflows. CircleCi runs `make format-check`,
-`make header-check` as part of our continuous integration. Pull requests should
-pass format-check and header-check without errors before being accepted.
+   * Implement the change
+   * If the change is large, post a preview Github pull request with the title prefixed with [WIP], and share with collaborators.
+   * Include tests and documentation as necessary.
 
-More details can be found at [.circleci/REAME.md](.circleci)
+1. Create a Github pull request (PR).
+   * Give the pull request a clear, brief description: when the pull request is merged, this will be retained in the extended commit message. Check out [How to Write Better Git Commit Messages – A Step-By-Step Guide](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/) and [How to Write a Git Commit Message](https://cbea.ms/git-commit/) to learn more about how to write good commit messages.
+   * Make sure the pull request passes the tests in CircleCI.
+   * If known, request a review from an expert in the area changed. If unknown, ask for help on Slack.
 
-## Contributor License Agreement ("CLA")
+1. Review is performed by one or more reviewers.
+   * This normally happens within a few days, but may take longer if the change is large, complex, or if a critical reviewer is unavailable. (feel free to ping the pull request).
+   * Address concerns and update the pull request.
 
-In order to accept your pull request, we need you to submit a CLA. You only need
-to do this once to work on any of Facebook's open source projects.
+1. After pushing the changes, add a comment to the pull-request, mentioning the
+   reviewers by name, stating the comments have been addressed. This is the only
+   way that a reviewer is notified that you are ready for the code to be reviewed
+   again.
 
-Complete your CLA here: <https://code.facebook.com/cla>
+1. Go to step 7.
```

Comment:
All of these have number 1, which is step 7 ?

Reply:
Markdown will convert it to sequential numbers.  But yeah it would be nice to make it correct even in plain text.

Reply:
You can use a preview feature to see how the document is rendered:

<img width="1039" alt="Screen Shot 2022-08-23 at 4 25 28 PM" src="https://user-images.githubusercontent.com/27965151/186258885-be739c86-12e2-49ed-a759-b54e8a6de10f.png">

Reply:
Dint know you could do that!

Reply:
> Markdown will convert it to sequential numbers. But yeah it would be nice to make it correct even in plain text.

I intentionally use 1. Makes it easy to add / remove items without having to fix all the numbers.